### DISCUSSION TITLE
How to reference metadata from _quarto.yml in a Quarto extension _extension.yml?

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#3987] [Q&A]
[MWE in github](https://github.com/fredguth/quarto-mwe-2)

Inside my extension, I want `_quarto.yml` to hide it is using a `book` project type under the hood.

Ideally, I would have in my `_quarto.yml`:

```_quarto.yml
project:
  type: tmp
tmp:
  tmp_title: "My Title"
```

I need to reference `$tmp.tmp_title$ ` inside `_extensions/tmp/_extension.yml`:
```_extensions/tmp/_extension.yml
title: tmp
author: Fred
version: 0.1.0
contributes:
  project:
    project:
      type: book
      output-dir: docs
    tmp:
      tmp_title: "title"
    book:
      title: ${tmp.tmp_title}
      author: "Jane Doe"
      date: "1/16/2023"
      chapters:
        - index.qmd
        - intro.qmd
        - summary.qmd
        - references.qmd
   ```

I tried to do it like I do inside any partials tex file, but it does not work in the .yml file.  How do I do that?

**Comment by dragonstyle** [comment]
You should be able to use the meta shortcode:

`quarto.yml`:
```markdown
project:
  type: tmp
tmp:
  tmp_title: "My Title"
bibliography: references.bib
```

`extension.yml`:
```markdown
title: tmp
author: Fred
version: 0.1.0
contributes:
  project:
    project:
      type: book
      output-dir: docs
    book:
      title: "{{< meta tmp.tmp_title >}}"
      author: "Jane Doe"
      date: "1/16/2023"
      chapters:
        - index.qmd
        - intro.qmd
        - summary.qmd
        - references.qmd
  formats:
    html:
      theme: cosmo
    pdf:
      documentclass: scrreprt


  # formats:
  #   pdf:
  #     template: monashthesis.tex
```


Would that work?

**Reply by fredguth** [reply]
I had already tried that.  But the output file name is weird:
```
Output created: docs/{{--meta-tmp.tmp_title--}}.pdf
```

But now... under closer look, I noticed that although it creates the document with this strange name, it actually updates the pdf correctly.

**Reply by fredguth** [reply]
Note: 
❯ quarto --version
1.2.313

**Reply by dragonstyle** [reply]
Ahh, I hadn't anticipated that (the file name). We resolve shortcodes like meta for the contents of documents, but they are not always resolved early enough in the process for all the various places they're used (outside of contents) to get the correct resolved value.

**Reply by fredguth** [reply]
Maybe I could write a lua filter that changes the name afterwards? 
Would that work?

How can I change the filename as a last step?

**Reply by dragonstyle** [reply]
You'd have to change the filename using a post render script, I think (just rename the file I think). The output filename is determined before we resolve the shortcodes, so it will be used 'as is' (as you noticed).

**Reply by fredguth** [reply]
Hmmm, I have never done that....
I just read the documentation.
I understand that I could add to my _extension.yml something like:
```_extensions/tmp/_extension.yml
title: tmp
author: Fred
version: 0.2.0
contributes:
  project:
    project:
      type: book
      output-dir: docs
      post-render: rename.py
    tmp:
      tmp_title: "title"
    book:
      title: "{{< meta tmp.tmp_title >}}"
...
   ```

But how do `rename.py` access `tmp.tmp_title` or `book.title`?  I see there is a way to access the files, will I have to parse the yaml files again in python?  Would it be easier in typescript?
